torchtune/modules/transformer.py

```diff
@@ -338,6 +339,7 @@ def __init__(
         self.num_heads = num_heads
         self.head_dim = head_dim
         self.causal_mask = None
+        self.cur_pos = None
```

Comment:
Can you explain what this is about?

Reply:
This tracks position for the TransformerDecoder since the layers also track it during decoding. This is already obsolete as Salaman is updating how cacheing handles position. But I'll leave this in until that is updated. This is basically a fix for the previous update around input_pos and kvcache.